libs/cart/driver/in-memory/src/backend/cart-shipping-methods/cart-shipping-methods.service.spec.ts

```diff
@@ -66,12 +69,49 @@ describe('DaffInMemoryBackendCartShippingMethodsService', () => {
 
     beforeEach(() => {
       reqInfoStub.url = cartUrl;
+    });
+
+    describe('when the cart has a shipping address', () => {
+      beforeEach(() => {
+        mockCart.shipping_address = cartAddressFactory.create();
+      });
+
+      describe('and when the cart already has available shipping methods', () => {
+        beforeEach(() => {
+          mockCart.available_shipping_methods = [mockCartShippingMethod];
+
+          result = service.get(reqInfoStub);
+        });
+
+        it('should return the cart\'s available shipping methods', () => {
+          expect(result.body).toEqual([mockCartShippingMethod]);
+        });
+      });
 
-      result = service.get(reqInfoStub);
+      describe('and when the cart has no available shipping methods', () => {
```

Comment:
for the second level of describes, the format is "and the cart has no..." instead of "and when the cart has no..." since the second "when" isn't needed => "when the cart has a shipping address, and the cart has no available shipping methods, it should initialize and return the cart's available shipping methods".

Reply:
removed